File: Computer-Seekho-.NET/Computer_Seekho_DN/Models/Image.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Computer_Seekho_DN.Models;

[Table("images")]
[Index("AlbumId", Name = "FK724cv7ds8vwsp7mpi6e5s7keq")]
public partial class Image
{
    [Key]
    [Column("image_id")]
    public int ImageId { get; set; }

    [Column("image_url")]
    [StringLength(255)]
    public string ImageUrl { get; set; } = null!;

    [Column("album_id")]
    public int? AlbumId { get; set; }

    [Column("image_description")]
    [StringLength(255)]
    public string ImageDescription { get; set; } = null!;

    [Column("image_title")]
    [StringLength(255)]
    public string ImageTitle { get; set; } = null!;

    [ForeignKey("AlbumId")]
    [InverseProperty("Images")]
    public virtual Album? Album { get; set; }
}

File: Computer-Seekho-Java/ComputerSeekho/src/main/java/com/Project/Entities/Image.java
package com.Project.Entities;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "images")
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "image_id")
    private int imageId;

    @Column(name = "image_url", length = 255, nullable = false)
    private String imageUrl;
    @ManyToOne
    @JoinColumn(name = "album_id", referencedColumnName = "album_id")
    private Album album;

     @Column(name = "image_title", length  = 255, nullable = false)
    private String imageTitle;

    @Column(name = "image_description", length = 255, nullable = false)
    private String imageDescrption;


    // Getters and Setters
    public int getImageId() {
        return imageId;
    }

    public void setImageId(int imageId) {
        this.imageId = imageId;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }
}
